File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:

        #hm = {"a": 0, "e": 0, "i": 0, "o": 0, "u": 0}
        hs = {'a','e','i','o','u'}
        firstWindow = s[0:k]
        c = 0
        maxi = 0
        for i in firstWindow:
            if i in hs:
                c+=1
        i = 0

        maxi = max(maxi, c)
        for j in range(k,len(s)):
            if s[i] in hs:
                c-=1
            i+=1
            if s[j] in hs:
                c+=1
            maxi = max(maxi, c)
        print(maxi)
        return maxi

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        
        HashSet<Character> hs = new HashSet<>();
        hs.add('a');
        hs.add('e');
        hs.add('i');
        hs.add('o');
        hs.add('u');
        int c=0;
        int res = 0;
        for(int i=0;i<k;i++){
            if(hs.contains(s.charAt(i))){
                c++;
            }
        }
        res = Math.max(res, c);

        for(int j=k;j<s.length();j++){
            if(hs.contains(s.charAt(j-k))){
                c--;

            }
            if(hs.contains(s.charAt(j))){
                c++;
            }
            res = Math.max(res, c);
        }


        return res;



    }
}